src/interface/storage.thrift

```diff
@@ -310,6 +310,61 @@ struct UpdateEdgeRequest {
     7: bool insertable,
 }
 
+struct ScanEdgeRequest {
+    1: common.GraphSpaceID space_id,
+    2: common.PartitionID part_id,
+    // start key of this block
+    3: optional binary cursor,
+    // If specified, only return specified columns, if not, no columns are returned
+    4: map<common.EdgeType, list<PropDef>> (cpp.template = "std::unordered_map") return_columns,
+    5: bool all_columns,
+    // max row count of edge in this response
+    6: i32 limit,
+    7: i64 start_time,
+    8: i64 end_time,
+}
+
+struct ScanEdgeResponse {
+    1: required ResponseCommon result,
+    2: map<common.EdgeType, common.Schema> (cpp.template = "std::unordered_map") edge_schema,
+    3: map<common.VertexID, list<ScanEdge>> (cpp.template = "std::unordered_map") edge_data,
```

Comment:
Why not return list directly

Reply:
In the commits before, I return a list directly, we can save some space with a map. WDYT?